models/ArticleManager.php

```diff
@@ -0,0 +1,395 @@
+            
+            $post   = new Article();
+            $post   ->setId($row['id']);
+            $post   ->setTitle($row['title']);
+            $post   ->setChapo($row['chapo']);
+            $post   ->setContent($row['content']);
+            $post   ->setDate($row['date']);
+            $post   ->setImage($row['image']);
+                        
+            $posts[] = $post;
+        }
+        return $posts;
+    }
+    
+     public function get_posts_blog_all(){
+        $req = $this->db->query("
+            SELECT  articles.id,
+                    articles.title,
+                    articles.chapo,
+                    articles.image,
+                    articles.date,
+                    articles.content,
+                    admins.name
+            FROM articles
+            JOIN admins
+            ON articles.writer=admins.email
+            WHERE posted='1'
+            ORDER BY date DESC
+        ");
+        
+        while($row = $req->fetch()){
+            
+            $post   = new Article();
+            $post   ->setId($row['id']);
+            $post   ->setTitle($row['title']);
+            $post   ->setChapo($row['chapo']);
+            $post   ->setContent($row['content']);
+            $post   ->setDate($row['date']);
+            $post   ->setImage($row['image']);
+                        
+            $posts[] = $post;
+        }
+        return $posts;
+    }
+    
+    // Fonction qui récupère un article avec le posted = 1
+    public function get_article_blog(){
+        $req = $this->db->query("
+            SELECT  articles.id,
+                    articles.title,
+                    articles.chapo,
+                    articles.image,
+                    articles.content,
+                    articles.date,
+                    admins.name
+            FROM articles
+            JOIN admins
+            ON articles.writer = admins.email
+            WHERE articles.id='{$_GET['id']}'
+            AND articles.posted = '1'
+        ");
+    
+        $post = [];
+        while($row = $req->fetch()){
+            
+            $post   = new Article();
+            $post   ->setId($row['id']);
+            $post   ->setTitle($row['title']);
+            $post   ->setChapo($row['chapo']);
+            $post   ->setContent($row['content']);
+            $post   ->setDate($row['date']);
+            $post   ->setImage($row['image']);
+                        
+            $posts[] = $post;
+        }
+        return $post;
+    }
+
+
+//FONCTIONS POUR LE BACKEND
+    // Fonction permettant de récupérer l'ensemble des articles
+    function get_posts(){
+        $req = $this->db->query("SELECT * FROM articles ORDER BY date DESC");
+        
+        $results = [];
+        while($rows = $req->fetchObject()){
+            $results[] = $rows;
+        }
+        
+        return $results;
+    }
+
+// Fonction permettant d'obtenir un article en particulier
+    function get_post(){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [get_post accesses the super-global variable $_GET.](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5379676)